backend/src/services/promptService.ts

```diff
@@ -5,17 +5,63 @@ const presets = JSON.parse(
     fs.readFileSync('files/presets.json').toString()
 ) as Preset[];
 
-const defaultPreset: Preset = {
-    name: 'default',
-    parameters: {
-        model: 'text-davinci-002',
-        prompt: '',
-        temperature: 0.5,
-        max_tokens: 300,
-        top_p: 0.999,
-        frequency_penalty: 0.52,
-        presence_penalty: 0.5,
-    },
+/**
+ * Converts our own parameters into GPT-approved parameters according to discussed upon factors
+ *
+ * @returns custom preset with correctly converted parameters
+ */
+
+const convertParameters = (
+    creativity: number,
+    quality: number,
+    inputLength: number,
+    outputLength: number
+): Preset => {
+    //cut the input length, normally between 1024-8000,
+    //to 4000 if quality 1-3, also halves max_tokens
+    let cutInputLength = inputLength;
+    let maxTokenFactor = 1;
+    if (quality < 4) {
+        if (inputLength > 4000) {
+            cutInputLength = 4000;
+        }
+        maxTokenFactor = 0.5;
+    }
+
+    const temperature = creativity * (4 / 5) + 0.2;
+    const presence_penalty = creativity * (3 / 5);
+    const frequency_penalty = creativity * (1 / 2);
```

Comment:
If there are constants, we can calculate them before and not do any unnecessary calculations, for example here:
4/5 = 0.8
3/5 = 0.6
1/2 = 0.5

Reply:
I disagree, the cost of float arithmetic is incredibly small and the constants are simple enough that I think we can keep them online (if they were complex constants like PI or other magic numbers, I would agree)